### ISSUE TITLE
Mean subtraction after cropping

### ISSUE THREAD

**Issue by mgarbade** [DrSleep/tensorflow-deeplab-resnet#25]
There is a bug in the [image transformation](https://github.com/DrSleep/tensorflow-deeplab-resnet/blob/master/deeplab_resnet/image_reader.py#L64). The mean subtraction must happen before the zero padding performed by `tf.image.resize_image_with_crop_or_pad`.

I'm trying to squeeze the functions
        img = tf.random_crop(img, [h, w, 3])
        label = tf.random_crop(label, [h, w, 1])
in there as well to get a random instead of a center crop, but I'm not sure if they take the same crop from image and label if I do it like this... So I keep looking into that.

**Comment by arslan-chaudhry** [comment]
Yes, this looks like a bug.
As for random crop, you can change the code to check if image dimensions are greater than '321', then do a random crop using _tf.random_crop_ , else pad the inputs accordingly. I am also doing the same thing, as always doing a centre-crop effectively reduces the amount of training data.

**Comment by DrSleep** [comment]
@mgarbade, nice catch! Fixed!

The `tf.random_crop` functions has a `seed` argument that you can provide to both instantiations to guarantee that the crop is the same for the image and the mask.
